### ISSUE TITLE
Scoping or filtering to specific namespace/pods

### ISSUE THREAD

**Issue by GoldenMouse** [logdna/logdna-agent-v2#163]
Currently is there any way to scope to a specific namespace and/or annotations such that only logs from these are forwarded. I don't see anything of the sort in the helm values.yaml.

**Comment by jorgebay** [comment]
Hi @GoldenMouse,
You should use `LOGDNA_EXCLUSION_RULES` env variable to set the exclusion rule to anything that does NOT match your namespace:

```yaml
- name: LOGDNA_EXCLUSION_RULES
  value: "!(*my-namespace*)"
```

The agent uses k8s logs symlinked to `/var/log/containers/`. K8S logging drivers will create symlink files to using the following pattern: `{pod-name}_{namespace}_{containerid}.log`. So you could use `"!(*my-namespace*)"` or `"!(*_my-namespace_*)"`

TLDR: We have an internal ticket to document this sort of allowlisting/denylisting.

**Comment by GoldenMouse** [comment]
Documentation would be good; this took me hours to figure out.  And still it doesn't work the way I want. Instead of having a bunch of exclusion rules, I tried to use inclusion rules. For instance, I only want a few namespace logged. Inclusion rules doesn't seem to work - it includes /var/log/ no matter what.

**Comment by jorgebay** [comment]
I agree we should support it: I've filed an internal ticket to support excluding or including only pods matching labels and annotations (LOG-9906).

About including `/var/log` no matter what, I also found it unexpected initially. We have an internal ticket (LOG-9352) to allow overriding that behaviour. In the meantime, we will tackle that issue with documentation.

To workaround this issue, the best solution is to use negative exclusion rules (exclude everything that doesn't match).

**Comment by Mmmariusz** [comment]
It is not working tested for 3.7 and 3.6.
```
          - name: LOGDNA_EXCLUSION_RULES
            value: '!(*default*)'
```
With this ^ rule I still see all logs.

**Comment by stsantilena** [comment]
Thank you for the feedback @Mmmariusz. We were able to reproduce this behavior and have filed an internal ticket to investigate further.

In the meantime, as a workaround, you could just use the `LOGDNA_EXCLUSION_RULES` to explicitly exclude log file you don't want to see.

Additionally, the above mentioned feature (LOG-9906) is complete and will be included in our next release. This will allow you to explicitly target Kubernetes objects such as namespaces, rather than using glob patterns.

**Comment by ranesagar** [comment]
> TLDR: We have an internal ticket to document this sort of allowlisting/denylisting.

@jorgebay - Was this ever released? If yes, could you please point to the documentation?

**Comment by jorgebay** [comment]
Hi Sagar,
I'm currently not dedicated to this project, I think the feature is documented under the K8s docs: https://github.com/logdna/logdna-agent-v2/blob/3.8.6/docs/KUBERNETES.md#configuration-for-kubernetes-metadata-filtering

The settings are `LOGDNA_K8S_METADATA_LINE_INCLUSION` and `LOGDNA_K8S_METADATA_LINE_EXCLUSION`.

@stsantilena Can you confirm that this is included in the latest 3.8 release? can this ticket be closed?

**Comment by stsantilena** [comment]
Hey @jorgebay - Yes, this work was included in the 3.8 release. I'll defer to @gjanco on whether or not he wants to close this issue as I'm no longer dedicated to this project either.

**Comment by gjanco** [comment]
Closing the loop on this.  This fix is currently available in all distros.
cc: @AdamHattrell
